src/sentry/integrations/perforce/client.py

```diff
@@ -73,16 +132,128 @@ def check_file(self, repo: Repository, path: str, version: str | None) -> object
+        else:
+            depot_root = repo.config.get("depot_path", repo.name).rstrip("/")
+
+            # Normalize depot_root to ensure it starts with //
+            if not depot_root.startswith("//"):
+                depot_root = f"//{depot_root}"
+
+            # Strip depot name from path if it duplicates depot_root
+            # e.g., depot_root="//depot", path="depot/app/file.py" → "app/file.py"
+            depot_name = depot_root.lstrip("/")  # "depot"
+            if path_without_rev.startswith(depot_name + "/") or path_without_rev == depot_name:
+                path_without_rev = path_without_rev[len(depot_name) :].lstrip("/")
+
+            # Remove leading slashes from relative path
+            path_without_rev = path_without_rev.lstrip("/")
+
+            # Handle Perforce streams: insert stream after depot
+            # Format: //depot/stream/path/to/file
+            if stream:
+                full_path = f"{depot_root}/{stream}/{path_without_rev}"
+            else:
+                full_path = f"{depot_root}/{path_without_rev}"
+
+        # Add file revision back if present
+        if revision:
+            full_path = f"{full_path}#{revision}"
+
+        return full_path
 
     def get_depots(self) -> list[dict[str, Any]]:
         """
         List all depots accessible to the user.
 
+        Uses p4 depots command to display a list of all depots.
+        API docs: https://www.perforce.com/manuals/cmdref/Content/CmdRef/p4_depots.html
+
         Returns:
             List of depot info dictionaries
         """
-        return []
+        p4 = self._connect()
+        try:
+            depots = p4.run("depots")
+            return [
+                {
+                    "name": depot.get("name"),
+                    "type": depot.get("type"),
+                    "description": depot.get("desc", ""),
+                }
+                for depot in depots
+            ]
+        finally:
+            self._disconnect(p4)
+
+    def get_user(self, username: str) -> dict[str, Any] | None:
+        """
+        Get user information from Perforce.
+
+        Uses p4 user command to fetch user details including email and full name.
+        API docs: https://www.perforce.com/manuals/cmdref/Content/CmdRef/p4_user.html
+
+        Args:
+            username: Perforce username
+
+        Returns:
+            User info dictionary with Email and FullName fields, or None if not found
+        """
+        p4 = self._connect()
+        try:
+            result = p4.run("user", "-o", username)
+            if result and len(result) > 0:
+                user_info = result[0]
+                return {
+                    "email": user_info.get("Email", ""),
+                    "full_name": user_info.get("FullName", ""),
+                    "username": user_info.get("User", username),
+                }
+            return None
+        except self.P4Exception:
+            return None
```

Comment:
Is there any reason we shouldn't bubble up an exception to the caller? In some instances, we have behavior in tasks that need this to trigger retries for transient failures.